### DISCUSSION TITLE
Dynamically updated json_encoders in v1.8?

### DISCUSSION THREAD

**Discussion by umesh-timalsina** [pydantic/pydantic#2793] [Question]
I have a pydantic base class where I register json encoders for non-pydantic classes via decorators. It seems like the way BaseModels are constructed in v1.8, dynamically updated json_encoders will not be reflected in the class's `__config__`. The workaround I cameup for this is to override the class's json method and force update the config with the dynamic dict of json_encoders. Without the update, serialization of custom encoded classes fails for `v1.8` but this works for `v1.7`.

**Comment by PrettyWood** [comment]
Hi @umesh-timalsina 
Can you please share a code snippet with the expected behaviour and the before / after ?
Thanks 🙏
